Firmware/ARYTHMATIK_Euclid/ARYTHMATIK_Euclid.ino

```diff
@@ -327,69 +479,119 @@ void loop() {
   }
 
   //-----------------trigger detect, reset & output----------------------
-  rst_in = digitalRead(11);//external reset in
+  #ifdef UPSIDEDOWN
+  rst_in = FastGPIO::Pin<13>::isInputHigh();  //external reset
+  #else
+  rst_in = FastGPIO::Pin<11>::isInputHigh();  //external reset
+  #endif
   if (old_rst_in == 0 && rst_in == 1) {
     for (k = 0; k <= 5; k++) {
       playing_step[k] = 0;
       disp_refresh = 1;
     }
   }
-  trg_in = digitalRead(13);//external trigger in
-  if (old_trg_in == 0 && trg_in == 0 && gate_timer + 1000 <= millis()) {
-    void BPM();
-    debug = 1;
-  }
-  else if (old_trg_in == 0 && trg_in == 1) {
+  #ifdef UPSIDEDOWN
+  trg_in = FastGPIO::Pin<11>::isInputHigh();
+  #else
+  trg_in = FastGPIO::Pin<13>::isInputHigh();
+  #endif
+  
+  // no external trigger for more than 8 sec -> internal clock (not implemented yet)
+  if (old_trg_in == 0 && trg_in == 0 && gate_timer + 8000 <= millis()) {
+    //useInternalClock = true;
+    //debug = 1;
+    disp_refresh = 1;
+  } else if (old_trg_in == 0 && trg_in == 1) {
     gate_timer = millis();
-    digitalWrite(4, HIGH);
+    //useInternalClock = false;
+    FastGPIO::Pin<4>::setOutput(1);
     debug = 0;
     for (i = 0; i <= 5; i++) {
-      playing_step[i]++;      //When the trigger in, increment the step by 1.
+      playing_step[i]++;
       if (playing_step[i] >= limit[i]) {
-        playing_step[i] = 0;  //When the step limit is reached, the step is set back to 0.
+        playing_step[i] = 0;  // step limit is reached
+      }
+    }
+    #ifdef UPSIDEDOWN
```

Comment:
There is an opportunity to simplify these changes by using the `#ifdef UPSIDEDOWN` flag to assign each output to a variable and use that instead of checking the `UPSIDEDOWN` flag each time. For example, add this definition at the top of your script:
```c
#ifdef UPSIDEDOWN
#define OUTPUT1 FastGPIO::Pin<8>
#define LED1 FastGPIO::Pin<0>
...
#else
#define OUTPUT1 FastGPIO::Pin<5>
#define LED1 FastGPIO::Pin<14>
...
#endif
```
Then use them like this without needing to check `#ifdef UPSIDEDOWN`:
```c
switch (k) {
  case 0:  //CH1
    OUTPUT1::setOutput(1);
    LED1::setOutput(1);
    break;
  ...
}
```